TRV12-EasymytripBus.txt

mishra-umesh opened the pull request.

sourabhsaini11 commented:
@mishra-umesh,kindly provide all flows as mentioned in log-submission-ui tab along with IGM & RSF. Additionally the folder & file structure is also incorrect. Ref to this [link](https://github.com/ONDC-Official/verification-logs/tree/main/TRV12/Pramaan) for correct structure example.

sourabhsaini11 commented:
@mishra-umesh
## common issues
bap_id & bpp_id are not registered at gateway
milliseconds should be precise upto 3 decimal places in timestamp 
missing location.city in context
null values can't be used in the payload, please change those and provide valid values
vehicle info is incomplete
Missing search_3-on_search_3 and select_-on_select_2 pair api calls
SETTLEMENT_TYPE can't be empty

## confirm
message_id can't be same as init's message_id

## search
missing milliseconds in time.timestamp

## on_search
How can location.city value be empty
invalid attributes descriptor, gps in location found
additional attributes found in fulfillments.stops, please remove those
How can gps value possibly be GPS_COORDINATES_HERE
tag group VEHICLE_AVAIBALITY should be used instead of VEHICLE_AVAILABILITY
missing tag group VEHICLE_GRID
fulfillment.type should be in enum format
payments array is wrong, please correct it

## search(Layout)
missing stops.id
location and time are not required in fulfillment.stops

## on_search(Layout)
missing tag group VEHICLE_GRID, VEHICLE_AMENITIES & VEHICLE_AVAIBALITY
How can location be empty
payments array is wrong, please correct it
additional attributes found in fulfillments.stops, please remove those

## select
fulfillments.tags are sent additionally

## on_select
How can location.city value be empty
invalid attributes descriptor, gps in location found
please remove null values and provide a valid payload

## init
missing person.gender
invalid city.code "std:"
How can tax_id be empty

## confirm
How can location.city value be empty
invalid attributes descriptor, gps in location found
billing info mismatch with what was sent in /init call
invalid object found in billing obj
invalid attributes found in provider obj
additional tag TYPE found in SEAT_GRID

## on_confirm
invalid city.code "std:0"
invalid attributes descriptor, gps in location found
milliseconds should be precise upto 3 decimal places in created_at
updated_at can't be empty
quote.breakup is incomplete

sourabhsaini11 commented:
@mishra-umesh, please provide logs from a valid ONDC preprod network transaction

amitsinha07 commented:
@mishra-umesh 

## IGM Feedbacks

kindly submit your logs in proper format create a seprate json file for all the apis like issue, on_issue, issue_status and on_issue_status

sourabhsaini11 commented:
@mishra-umesh
## common_issue
Bap_id & Bpp_id are not subscribed to preprod registry. Again please only provide valid transaction logs genrated through a valid ONDC network transaction
How can **city.code** value be sent as "1059", also values keep
Why is **message_id** getting changed b/w same action on_action pair calls
How can milliseconds(000) in timestamp be same for all provided calls
**Empty values** can't be present
Missing search_1, on_search_1 calls

## search
vehicle info is incomplete

## on_search
invalid keys Context & Message found
**gps** in stops can't have empty values
**fulfillment.type** should be in an enum format
Missing **VEHICLE_GRID**
Please use a valid id at **item.id**
**payments** tags are incorrect

## search(Layout)
How can fulfillment.id be 4776413 if the same hasn't be sent before

## search(Layout)
How can item.id refer F1 when the same is not even present in contract

## select
Again fulfillment.id is changed to "2239"
Item.id 4896734 hasn't been sent by seller in past calls, can't be referred here

## on_select
How can F1 & F2 id's be present if BAP is not even selecting those id's.

Please get these issues sorted first and only provide valid transactional logs
Also Folder structure is also incorrect, can you explain what's the diff b/w Buyer & Seller folders.

sourabhsaini11 commented:
@mishra-umesh
## common_issues
missing search on_search pair calls (one with catalog of all fulfillments & one with search of specific bus service)
**vehicle.capacity** should be of type number   
**quote.breakup** summation doesn't match against **quote.price**
values found to be in incorrect format for **cancellation_terms.cancel_by**
incorrect value found for tag **SETTLEMENT_AMOUNT**
missing value for **location.gps**

## search
invalid info found for **vehicle object**, empty values can't be present & capacity can't be 0

## on_select
tag **NUMBER** value can't be empty
multiple fulfillments with type as **TICKET** found
tag **SEAT_PRICE** value & items.price are changed b/w on_select_1 & on_select_2

## confirm
missing **location.city**

## on_confirm
title **SEAT_FARE** value is changed from what was sent in on_init

## on_cancel
**created_at** should match on_confirm's **created_at**

mishra-umesh commented:
**common_issues**

1. As Per our current bus flow we have only one search where we search from one station code to another station code
2. we have changed the vehicle.capaicty to integer from string.
3. Quote.breakup is correct as per our logs,eg is down below
"quote": {
        "price": {
          "currency": "INR",
          "value": "1764.0"
        },
        "breakup": [
          {
            "title": "BASE_FARE",
            "price": {
              "currency": "INR",
              "value": "1680.0"
            },
            "item": {
              "id": "4922421",
              "price": {
                "currency": "INR",
                "value": "1680.0"
              },
              "quantity": {
                "selected": {
                  "count": 1
                }
              }
            }
          },
          {
            "title": "TAX",
            "price": {
              "currency": "INR",
              "value": "0"
            },
            "item": {
              "tags": [
                {
                  "descriptor": {
                    "code": "TAX"
                  },
                  "list": [
                    {
                      "descriptor": {
                        "code": "CGST"
                      },
                      "value": "0"
                    },
                    {
                      "descriptor": {
                        "code": "SGST"
                      },
                      "value": "0"
                    }
                  ]
                }
              ]
            }
          },
          {
            "title": "CONVENIENCE_FEE",
            "price": {
              "currency": "INR",
              "value": "84.0"
            }
          },
          {
            "title": "SEAT_FARE",
            "price": {
              "currency": "INR",
              "value": "1764.0"
            }
          }
        ]
      }
	  
	  base fare of the bus is 1680 and convenience fee is 84 summation equals to 1764.
	  if there is any mistake please let us know
	  
	  4. cancellation_terms.cancel_by- we have provided the durations by whihch user can cancel and under it we have provided the precentage
	     from the document we are unable to understand what is PT60M.
	  5. Settlement_Amount is the final booking amout.
	  6. we don't have gps location as we are using station codes.
	  
	  **Search**
	  vehicle object- when we are doing first search at that time we dont have the data of the vehicle.
	  
	  **on_select**
	  1.tag NUMBER- we dont show the seat number of the seat which is not available.
	  2. Multiple Tickets are the number of seats which we are showing as seat layout.
	  3. Seat_PRICE
	   on_select1 we are showing the minimum price of the all the seats present in the perticular bus
	   while in on_select2 we are showing the price of the selected seat.
	   **confirm**
	   We will add the missing location.city
	   **on_confirm**
	   we have fixed the issue and will add it
	   **on_cancel**
	   How can created_at for the cancel and confirm be same
	   on_cancel created at will the value at which the cancel request is made while created_at for the confrim is the value at which confimation occured.

92shreyansh commented:
> 1. As Per our current bus flow we have only one search where we search from one station code to another station code
Understandable for interoperatability your seller needs to support all search you may as BAP choose to directly use 3rd search(station code based). Also how do you plan to get all valid station code for any other BPP?

> 2. we have changed the vehicle.capaicty to integer from string.
This comes from beckn and needs to be a string

> 3. Quote.breakup is correct as per our logs,eg is down below
>    "quote": {
>    "price": {
>    "currency": "INR",
>    "value": "1764.0"
>    },
>    "breakup": [
>    {
>    "title": "BASE_FARE",
>    "price": {
>    "currency": "INR",
>    "value": "1680.0"
>    },
>    "item": {
>    "id": "4922421",
>    "price": {
>    "currency": "INR",
>    "value": "1680.0"
>    },
>    "quantity": {
>    "selected": {
>    "count": 1
>    }
>    }
>    }
>    },
>    {
>    "title": "TAX",
>    "price": {
>    "currency": "INR",
>    "value": "0"
>    },
>    "item": {
>    "tags": [
>    {
>    "descriptor": {
>    "code": "TAX"
>    },
>    "list": [
>    {
>    "descriptor": {
>    "code": "CGST"
>    },
>    "value": "0"
>    },
>    {
>    "descriptor": {
>    "code": "SGST"
>    },
>    "value": "0"
>    }
>    ]
>    }
>    ]
>    }
>    },
>    {
>    "title": "CONVENIENCE_FEE",
>    "price": {
>    "currency": "INR",
>    "value": "84.0"
>    }
>    },
>    {
>    "title": "SEAT_FARE",
>    "price": {
>    "currency": "INR",
>    "value": "1764.0"
>    }
>    }
>    ]
>    }
>    base fare of the bus is 1680 and convenience fee is 84 summation equals to 1764.
>    if there is any mistake please let us know
This is inaccurate as the sum of all break up should result into total quote value but I see the SEAT_FARE break up is not considered in case the seat price is embeeded in the base bare then mention SEAT_FARE is 0

>    4. cancellation_terms.cancel_by- we have provided the durations by whihch user can cancel and under it we have provided the precentage
>       from the document we are unable to understand what is PT60M.
The duration needs to follow ISO 8601 format the current format is not standard and my not be recived from all BPP as well not all BAPs would be able to process this as desired

>    5. Settlement_Amount is the final booking amout.
This will be final settlement amount that will move from collector to reciever eg. BAP collecting  then settlement amount = total order value - Buyer Finder Fees, BPP collecting then settlement amount = Buyer Finder Fees

>    6. we don't have gps location as we are using station codes.
gps here are the gps of stations/stops which is use full for BAP to build a experience 

>    **Search**
>    vehicle object- when we are doing first search at that time we dont have the data of the vehicle.
How would will the buyer app display the vehicle details along with eminites and availability
>    **on_select**
>    1.tag NUMBER- we dont show the seat number of the seat which is not available.
What should BAP select here?
>    2. Multiple Tickets are the number of seats which we are showing as seat layout.
This needs to be represented as different Ticket Fulfillments
>    3. Seat_PRICE
>    on_select1 we are showing the minimum price of the all the seats present in the perticular bus
>    while in on_select2 we are showing the price of the selected seat.
The price needs to be passed on to each seat so that customer can make a informed decision
>    **confirm**
>    We will add the missing location.city
>    **on_confirm**
>    we have fixed the issue and will add it
Okay
>    **on_cancel**
>    How can created_at for the cancel and confirm be same
>    on_cancel created at will the value at which the cancel request is made while created_at for the confrim is the value at which confimation occured.
The created_at is a order level field and denotes the created_at of order not of the API call, As the order is created during on_confirm then the created_at across all post order on_action should have the same value

amitsinha07 commented:
**@mishra-umesh**  

The flow for **IGM 2.0.0** appears to be incomplete. Kindly refer to the **IGM 2.0.0 Developer Guide** for the correct implementation.  

You can find the guide at the following link:  
[https://ondc-official.github.io/ONDC-NTS-Specifications](https://ondc-official.github.io/ONDC-NTS-Specifications)

JyotiSharma0711 commented:
Please proceed with Pramaan testing at https://pramaan.ondc.org/ for the pre-production environment. Once you obtain a clean report from Pramaan, simply upload it—there is no need for a GitHub PR log review for the IGM 2.0.0  use case.

Kindly note that this PR is scheduled to be closed within the next 1 to 2 weeks. Please ensure that any required changes from your side are addressed within this timeframe.

JyotiSharma0711 commented:
Reminder:

Kindly note that this PR is scheduled to be closed within the next 1 to 2 weeks. Please ensure that any required changes from your side are addressed within this timeframe.

JyotiSharma0711 commented:
Based on the last update, we are proceeding to close this PR. Thank you for your time and support
